string[] node1
string[] node2
float64[] costs
